src/controllers/team.controller.ts

```diff
@@ -49,3 +50,37 @@ export async function updateMembers(
       .json(new HttpException(HttpStatusCode.UNAUTHORIZED, "Only team host can update team members."));
   }
 }
+
+export async function addMemberByName(
+  req:RequestWithUser,
+  res:Response,
+): Promise<Response | void> {
+
+  const id: string = req.params.id;
+  const userToAdd: string = req.params.username;
+
+  const user = await userService.getByUsername(userToAdd);
+  const team = await teamService.getById(id);
+
+  if (!user) {
+    return res
+    .status(HttpStatusCode.NOT_FOUND)
+    .json(new HttpException(HttpStatusCode.NOT_FOUND, "User not found"));
+  }
+
+  if (!team) {
+    return res.status(HttpStatusCode.NOT_FOUND)
+    .json(new HttpException(HttpStatusCode.NOT_FOUND, "Team not found"));
+  }
+
+  if (!team.members.includes(userToAdd)) {
```

Comment:
Look at the documentation, please. Team's `members` array contains user IDs, not usernames.

Reply:
You need to check whether `!team.members.includes(user._id)` and `team.members.push(user._id)`